pkg/apiserver/rest/asset_scan_controller.go

```diff
@@ -83,14 +83,20 @@ func (s *ServerImpl) PatchAssetScansAssetScanID(ctx echo.Context, assetScanID mo
 	}
 
 	// check that an asset scan with that id exists.
-	_, err = s.dbHandler.AssetScansTable().GetAssetScan(assetScanID, models.GetAssetScansAssetScanIDParams{})
+	existingAssetScan, err := s.dbHandler.AssetScansTable().GetAssetScan(assetScanID, models.GetAssetScansAssetScanIDParams{})
 	if err != nil {
 		if errors.Is(err, gorm.ErrRecordNotFound) {
 			return sendError(ctx, http.StatusNotFound, fmt.Sprintf("asset scan was not found. assetScanID=%v: %v", assetScanID, err))
 		}
 		return sendError(ctx, http.StatusInternalServerError, fmt.Sprintf("failed to get asset scan. assetScanID=%v: %v", assetScanID, err))
 	}
 
+	// check for valid resource cleanup state transition
+	err = existingAssetScan.ResourceCleanup.UpdateState(*assetScan.ResourceCleanup.State)
```

Comment:
Need to check 
`if assetScan.ResourceCleanup != nil && assetScan.ResourceCleanup.State != nil`

This is a patch operation and the ResourceCleanup field will not always be set

Reply:
I'd probably add a getter to the `AssetScan` the same way as we have for other objects to get their fields safely.

```suggestion
	func (s *AssetScan) ResourceCleanup() (*ResourceCleanup, bool) {
	    if s.ResourceCleanup == nil || s.ResourceCleanup.State == nil {
	        nil, false
	    }
	    
	    return s.ResourceCleanup, true
	}
```

And use this method for checking whether the `ResourceCleanup` object is valid or not.

```suggestion
        if _, ok := existingAssetScan.ResourceCleanup(); !ok {
            return ...
        }
        
	err = existingAssetScan.ResourceCleanup.UpdateState(*assetScan.ResourceCleanup.State)
```